src/script/standard.cpp

```diff
@@ -263,6 +274,50 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::v
                         break;
                 }
             }
+            else if(opcode2 == OP_ADDRESS_TYPE)
+            {
+                try {
+                    uint64_t val = CScriptNum::vch_to_uint64(vch1);
+                    if(val < 1 && val > 5)
```

Comment:
Using constants for this might be more clear of the meaning of these numbers. Also if you attempt to use address type "2" for instance, will it be considered an invalid and non-standard tx?

Reply:
Added new enum `addresstype` for the constants